packages/language/test/document.js

```diff
@@ -0,0 +1,824 @@
+      AFFIX: 'Affix'
+    };
+
+    it('should define the correct parts of speech', function() {
+      assert.deepEqual(Document.PART_OF_SPEECH, expectedPartOfSpeech);
+    });
+  });
+
+  describe('annotate', function() {
+    it('should make the correct API request', function(done) {
+      document.request = function(grpcOpts, reqOpts) {
+        assert.deepEqual(grpcOpts, {
+          service: 'LanguageService',
+          method: 'annotateText'
+        });
+
+        assert.deepEqual(reqOpts, extend(
+          {
+            features: {
+              extractDocumentSentiment: true,
+              extractEntities: true,
+              extractSyntax: true
+            }
+          },
+          document.reqOpts
+        ));
+
+        done();
+      };
+
+      document.annotate(assert.ifError);
+    });
+
+    it('should allow specifying individual features', function(done) {
+      document.request = function(grpcOpts, reqOpts) {
+        assert.deepEqual(reqOpts.features, {
+          extractDocumentSentiment: false,
+          extractEntities: true,
+          extractSyntax: true
+        });
+
+        done();
+      };
+
+      document.annotate({
+        entities: true,
+        syntax: true
+      }, assert.ifError);
+    });
+
+    describe('error', function() {
+      var apiResponse = {};
+      var error = new Error('Error.');
+
+      beforeEach(function() {
+        document.request = function(grpcOpts, reqOpts, callback) {
+          callback(error, apiResponse);
+        };
+      });
+
+      it('should exec callback with error and API response', function(done) {
+        document.annotate(function(err, annotation, apiResponse_) {
+          assert.strictEqual(err, error);
+          assert.strictEqual(annotation, null);
+          assert.strictEqual(apiResponse_, apiResponse);
+          done();
+        });
+      });
+    });
+
+    describe('success', function() {
+      var apiResponses = {
+        default: {
+          language: 'EN',
+          sentences: [],
+          tokens: []
+        },
+
+        withSentiment: {
+          documentSentiment: {}
+        },
+
+        withEntities: {
+          entities: []
+        },
+
+        withSyntax: {
+          sentences: {},
+          tokens: {}
+        }
+      };
+
+      apiResponses.withAll = extend(
+        {},
+        apiResponses.default,
+        apiResponses.withSentiment,
+        apiResponses.withEntities,
+        apiResponses.withSyntax
+      );
+
+      function createRequestWithResponse(apiResponse) {
+        return function(grpcOpts, reqOpts, callback) {
+          callback(null, apiResponse, apiResponse);
+        };
+      }
+
+      beforeEach(function() {
+        Document.formatSentiment_ = util.noop;
+        Document.formatEntities_ = util.noop;
+        Document.formatSentences_ = util.noop;
+        Document.formatTokens_ = util.noop;
+      });
+
+      it('should return the language and syntax by default', function(done) {
```

Comment:
Ah, nevermind I see where you were going with this now. Could we separate these into different tests? I know it would be adding some extra verbosity, but I think if the tests were clearly labeled it would be easier to follow.